rotkehlchen/externalapis/etherscan.py

```diff
@@ -108,7 +108,10 @@ def __init__(
     ) -> None:
         super().__init__(database=database, service_name=service)
         self.msg_aggregator = msg_aggregator
+        self.original_service = service
+        self.cached_settings = CachedSettings()
         self.chain = chain
+        self.chain_id = self.chain.to_chain_id().serialize()
```

Comment:
Don't do this. Why would you save this in memory of each etherscan object? You only use it a single if condition of a single function. Just do it there.

The only thing that makes sense to be saved in memory along with each object is something that's used everywhere and a lot

Reply:
Removed this. It was [suggested](https://github.com/rotki/rotki/pull/9159#discussion_r1898032819) to do it like this earlier when the chain id was being referenced for every api call.